common/k8stest/util_rdma.go

```diff
@@ -41,3 +46,86 @@ func ListRdmaDevice(node string) ([]RdmaDeviceNetworkInterface, error) {
 	logf.Log.Info("RDMA device", "node", node, "list", rdmaDeiceList)
 	return rdmaDeiceList, nil
 }
+
+func CreateRdmaDeviceOnNode(node string) error {
+	rdmaDeiceList, err := ListRdmaDevice(node)
+	if err != nil {
+		return err
+	}
+	if len(rdmaDeiceList) == 0 {
+		logf.Log.Info("RDMA device not found", "node", node, "list", rdmaDeiceList)
+		//create rdma device
+		nodeIp, err := GetNodeIPAddress(node)
+		if err != nil {
+			return fmt.Errorf("failed to get node %s ip, error: %v", node, err)
+		}
+
+		// get interface name
+		iface := e2e_config.GetConfig().NetworkInterface
+		out, err := e2e_agent.CreateRdmaDevice(*nodeIp, RdmaDeviceName, iface)
+		if err != nil {
+			return err
+		}
+		logf.Log.Info("Device created", "node", node, "output", out, "interface", iface)
+
+	}
+	rdmaDeiceList, err = ListRdmaDevice(node)
+	if err != nil {
+		return err
+	}
+	logf.Log.Info("RDMA device", "node", node, "list", rdmaDeiceList)
+	return nil
+}
+
+func CreateRdmaDeviceOnAllWorkerNodes() error {
+	workerNodes, err := ListWorkerNode()
+	if err != nil {
+		return err
+	}
+	for _, node := range workerNodes {
+		err := CreateRdmaDeviceOnNode(node.NodeName)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func GetVolumeProtocol(volUuid string) (string, error) {
+	deviceUri, err := GetMsvDeviceUri(volUuid)
+	var protocolSubstring string
+	if err != nil {
+		return "", err
+	}
+	// deviceUri: nvmf+tcp://<some-random-string>
+	firstColonIndex := strings.Index(deviceUri, ":")
+	if firstColonIndex != -1 {
+		protocolSubstring = deviceUri[:firstColonIndex]
+	} else {
+		return "", fmt.Errorf("no colon found in the device uri %s", deviceUri)
+	}
+	logf.Log.Info("Device", "Uri", deviceUri, "Protocol substring", protocolSubstring)
+	return protocolSubstring, nil
+}
+
+func IsVolumeAccessibleOverRdma(volUuid string) (bool, error) {
```

Comment:
Can we have descriptions for these functions with examples?
For example it isn't clear why  the check for accessibility over tcp excludes  protcol strings that include rdma

Reply:
added comments https://github.com/openebs/openebs-e2e/compare/118dc74950762944ffddacd3484e7b7ba501adff..3bd4c1ade1b0a643ad5a288a5af53803d7538be9